### ISSUE TITLE
Windows 10 client drops connection after a while due to cipher suite mismatch

### ISSUE THREAD

**Issue by ddwo** [trailofbits/algo#316]
### OS / Environment

server: Ubuntu 16.04.2 LTS
client:  Windows 10 Version 1703 (10.0.15603)

### Summary of the problem

Windows will connect to the VPN correctly but after a period of time the connection drops and the connection needs to be restarted manually.

In the log file there's some encryption negotiation:
```Mar 30 04:22:07 02[NET] <ikev2-pubkey|1> received packet: from <CLIENT IP>[55084] to <SERVER IP>[4500] (304 bytes)
Mar 30 04:22:07 02[ENC] <ikev2-pubkey|1> parsed CREATE_CHILD_SA request 2 [ N(REKEY_SA) SA No TSi TSr ]
Mar 30 04:22:07 02[CFG] <ikev2-pubkey|1> received proposals: ESP:AES_CBC_256/HMAC_SHA2_256_128/NO_EXT_SEQ
Mar 30 04:22:07 02[CFG] <ikev2-pubkey|1> configured proposals: ESP:AES_GCM_16_128/ECP_256/NO_EXT_SEQ, ESP:AES_CBC_256/HMAC_SHA2_256_128/MODP_2048/NO_EXT_SEQ
Mar 30 04:22:07 02[IKE] <ikev2-pubkey|1> no acceptable proposal found
Mar 30 04:22:07 02[IKE] <ikev2-pubkey|1> failed to establish CHILD_SA, keeping IKE_SA
```

I'm not seeing the default cipher suite referenced at https://github.com/trailofbits/algo/issues/9#issuecomment-243264378, and the `Get-VpnConnection` powershell command on the client reports a "Custom" encryption level, so I think the generated ps1 script is working correctly.

I'm able to change the cipher suite in `/etc/ipsec.conf` to match what the windows client requests in the charon log, and it does work, but I'm not exactly sure if I should be doing that.

### The way of deployment (cloud or local)

local

### Full log

[charon-bad.log.txt](https://github.com/trailofbits/algo/files/883139/charon-bad.log.txt)

**Comment by dguido** [comment]
You need to choose Windows support when you build the server. It changes the default ciphers. You also need to configure the local adapter to support those ciphers with Powershell. Please see the readme.

**Comment by ddwo** [comment]
I did. I moved ``ipsec.conf`` and reran ``algo`` and definitely selected the windows option and the regenerated ``ipsec.conf`` was identical. I can look at the [ipsec.conf.j2](https://github.com/trailofbits/algo/blob/ecb6b498b9a4edff16156e31e244e4f1408d7eff/roles/vpn/templates/ipsec.conf.j2) template and the generated ``ipsec.conf`` file and see that the windows 10 portion of the template was inserted correctly. I'm not sure how to double-check the ciphers on the windows side but as I said, I'm able to connect to the vpn until it tries to negotiate the esp cipher, and windows reports a "custom" encryption level.
